### DISCUSSION TITLE
Commented-out Camera from Config, its Recordings now Lost Forever?

### DISCUSSION THREAD

**Discussion by teksrq** [blakeblackshear/frigate#8940] [General]
Had to take a camera down from a wall that was being repaired; I was planning to reinstall it, so I figured I'd just pound out all its lines in the config for that camera (made sense at the time). Didn't intend to remove/lose all the stored recordings for it, but I also didn't think to check the docs for the proper process of disabling a camera ("***enabled: false***" in that particular camera config was the move, in hindsight). Now all the recordings for that camera are gone, thumbnails and events are still present, but clicking on any event is a 404 file not found. Looking in the storage folders for each day, the folder for that particular camera, "sideyard",  is missing in all folder days. 

I used Testdisk/PhotoRec to recover thousands of the lost mp4s, but they are all unorganized and missing EXIF data that might allow me to figure out what day/time/name of the original. Is there any hope of returning all those folders/files to their original locations? It's a 4TB drive with only 400GB of storage being used, so I'm pretty positive the data still exists.

**Comment by NickM-27** [comment]
this is expected, frigate will clean up recordings for cameras that have been removed from the config. I wouldn't see any way of getting the files back because even if the files are there you'd have to manually replace every DB entry

**Reply by teksrq** [reply]
Thanks, that's what I was figuring, but all the thumbnails and event entries still present gave me hope. Why wouldn't those clean/clear out in the same process? 

Is there a quick way to remove all the thumbnails. event and recording entries for that camera? Guessing I can just change the settings for that camera in the config, removing the retention settings.

Could garbage collection be a configurable time feature? I lost several months of video in less than 24 hrs with TBs of storage free on the drive (all my fault, of course). Even knowing the configuration option exists might save someone(s) from making the mistake I did....

**Reply by NickM-27** [reply]
> but all the thumbnails and event entries still present gave me hope. Why wouldn't those clean/clear out in the same process?

they will be, recordings and events are cleaned up as different processes and their cleanup interval is different. When a camera is removed from the config the default retention period is used which is 10 days for snapshots

> Is there a quick way to remove all the thumbnails. event and recording entries for that camera? Guessing I can just change the settings for that camera in the config, removing the retention settings.

manually using sqlite with the database otherwise just waiting

> Could garbage collection be a configurable time feature? I lost several months of video in less than 24 hrs with TBs of storage free on the drive (all my fault, of course). Even knowing the configuration option exists might save someone(s) from making the mistake I did....

it is configurable

**Comment by leccelecce** [comment]
I think this is quite dangerous behaviour - it's easy to see how a simple misconfiguration (in a tool without a GUI config editor) could result in irrecoverable camera footage loss. Couldn't this also be triggered by something as simple as someone renaming a camera?

Rather than the storage maintainer automatically deleting data for cameras it doesn't find in config, how about it a) logs a warning when it detects stored recordings without a matching camera in config and b) show them as "Obsolete Recordings" on the Storage page, with a Delete button so the user can make an explicit choice about whether they want to lose the footage?

**Reply by NickM-27** [reply]
> Rather than the storage maintainer automatically deleting data for cameras it doesn't find in config

that is not what it does. When a camera is not found in the config it uses the global recording config. If the user has not set a global recording config then it uses the default values which includes 0 days of 24/7 recording.

**Reply by leccelecce** [reply]
That still feels like a potential footgun for anyone who specifies retention per-camera but not globally, or (more likely) have a global figure that's relatively low e.g. 7 days but a higher value for certain "high value" cameras e.g. 28 days - you could rename a camera, then discover the next day you've lost 3 out of 4 weeks of footage.

At minimum I think there should be something in the docs that highlights the risk of renaming cameras or temporarily removing them from the config, will send a PR.

**Reply by teksrq** [reply]
Footgun ->* me. My per camera retention settings are 120 days; in all the example configs I perused, I never saw/noticed a global retention setting. Since I was only using < 400GB of the 4TB drive, I made an assumption that the data wouldn't be wiped (why do garbage collection when the can is no where close to full). Coming from over a decade of using another *nix-based NVR where I had to sometimes remove old camera data from the command line, losing all recordings in less than 24 hrs was a shock. 

Completely my negligence, but I agree that some flashing-red notice to inform users would surely save some people the trouble of learning the hard way.
